crates/base_db/src/input.rs

```diff
@@ -169,7 +169,9 @@ pub struct ProcMacro {
 impl Eq for ProcMacro {}
 impl PartialEq for ProcMacro {
     fn eq(&self, other: &ProcMacro) -> bool {
-        self.name == other.name && Arc::ptr_eq(&self.expander, &other.expander)
+        let thisptr = self.expander.as_ref() as *const dyn ProcMacroExpander as *const u8;
+        let otherptr = other.expander.as_ref() as *const dyn ProcMacroExpander as *const u8;
+        self.name == other.name && std::ptr::eq(thisptr, otherptr)
```

Comment:
oh dear... https://github.com/rust-lang/rust/pull/80505

Reply:
There's a lot of nasty in pointer comparisons here I guess, though once you start transmuting stuff I think I tap out on knowing what's sane 😬 

Do you mean, by this comment, that you want this particular change dropped, annotated that at some point it might be revertable, or do you just mean to mention the horror :grin: ?